pandas/tests/plotting/test_hist_method.py

```diff
@@ -595,3 +625,18 @@ def test_axis_share_xy(self):
 
         assert ax1._shared_y_axes.joined(ax1, ax2)
         assert ax2._shared_y_axes.joined(ax1, ax2)
+
+    @pytest.mark.parametrize(
+        "histtype, expected",
+        [
+            ("bar", True),
+            ("barstacked", True),
+            ("step", False),
+            ("stepfilled", True),
+        ],
+    )
+    def test_histtype_argument(self, histtype, expected):
```

Comment:
same here need renaming. would be nicer if you can merge this test with the one above by adding a `by` parameterization
something like
```python
@pytest.mark.parametrize("by", [None, "a"])
@pytest.mark.parametrize(
        "histtype, expected",
        [
            ("bar", True),
            ("barstacked", True),
            ("step", False),
            ("stepfilled", True),
        ],
    )
```

Reply:
but i don't have strong opinion on it, also fine if you leave it as is. i brought it up just because I recalled there were some cleaning before to remove those giant classes in tests files.

Reply:
I tried this, but since there are dependencies to TestPlotBase (both _check_filling and _flatten_visible ) we need to add something like shown below to connect it to the old test logic in classes. IMO this looks less elegant than the current imp, so my preference would be to keep it as is.
```python
tp = TestPlotBase()
tp._check_filling(ax, filled=expected)
```

Reply:
ehh, okay, then it's fine, i forgot that it is a base class for those shared common methods. 

my bad, sorry